torch/csrc/autograd/engine.h

```diff
@@ -152,7 +166,7 @@ struct TORCH_API Engine {
   // for the graph. This API should only be used by internal autograd specific
   // machinery and shouldn't be exposed to users in anyway.
   variable_list execute_with_graph_task(
-      GraphTask& graph_task,
+      std::shared_ptr<GraphTask> graph_task,
```

Comment:
If this isn't actually going to take ownership, you should pass this by reference to avoid refcount bump